netty/src/main/java/io/grpc/netty/CancelClientStreamCommand.java

```diff
@@ -18,25 +18,27 @@
 
 import com.google.common.base.Preconditions;
 import io.grpc.Status;
+import javax.annotation.Nullable;
 
 /**
  * Command sent from a Netty client stream to the handler to cancel the stream.
  */
 class CancelClientStreamCommand extends WriteQueue.AbstractQueuedCommand {
   private final NettyClientStream.TransportState stream;
-  private final Status reason;
+  @Nullable private final Status reason;
 
   CancelClientStreamCommand(NettyClientStream.TransportState stream, Status reason) {
     this.stream = Preconditions.checkNotNull(stream, "stream");
-    Preconditions.checkNotNull(reason, "reason");
-    Preconditions.checkArgument(!reason.isOk(), "Should not cancel with OK status");
+    Preconditions.checkArgument(
```

Comment:
Im a little uncomfortable with using null here.   I think a sentinel value like Status.CANCELED.withDescription("early trailers")   would be a safer approach.   Anything else that accesses the reason may expect it to be non null.    Also, most of gRPC makes the status non null.

Reply:
The point of null here is to be able to send RST_STREAM to the server without invoking `transportReportStatus` (again) in `NettyClientHandler#cancelStream`. This field is accessed in exactly one place (`NettyClientHandler#cancelStream`, which should be the only thing parsing these commands anyways?), and it checks for null to decide whether to report the status to the transport. I could accomplish the same by adding a `RstStreamCommand` (and had this in an earlier draft of this PR), but it would be functionally almost identical to `CancelClientStreamCommand` so seemed like unnecessary redundancy.